tensorrt_llm/llmapi/llm_args.py

```
@@ -2181,6 +2191,32 @@ def validate_max_util_for_resume(cls, v: float):
+        """Auto-detect unified memory and adjust KV cache defaults.
+
+        On unified memory systems (e.g. Grace Blackwell / DGX Spark), CPU and GPU
+        share the same physical LPDDR5x pool. KV cache offload is a no-op and the
+        host_cache_size budget should be folded into the primary GPU pool. The C++
+        runtime handles the actual optimization transparently; this validator logs
+        the detection and adjusts Python-level defaults for clarity.
+        """
+        if self.enable_unified_memory_optimization is None:
+            try:
+                unified = is_device_integrated()
+            except Exception:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)